docs/source/developer_guide/benchmarking.md

```diff
@@ -0,0 +1,141 @@
+# Benchmarking cuDF
+
+The goal of the benchmarks in this repository is to measure the performance of various cuDF APIs.
+Benchmarks in cuDF are written using the
+[`pytest-benchmark`](https://pytest-benchmark.readthedocs.io/en/latest/index.html) plugin to the
+[`pytest`](https://docs.pytest.org/en/latest/) Python testing framework.
+Using `pytest-benchmark` provides a seamless experience for developers familiar with `pytest`.
+We include benchmarks of both public APIs and internal functions.
```

Comment:
I don't think this PR has internal benchmarks. But still it's nice to include this line for future guidance.

Reply:
Yeah I leave a few carrots in here as stuff to do that should be prioritized probably. I'll look at making issues for them.